lib/pgBackRest/Manifest.pm

```diff
@@ -556,6 +556,177 @@ sub isTargetTablespace
+# checkDeltaFile
+#
+# Determine if the delta option should be enabled. Only called if delta has not yet been enabled.
+####################################################################################################################################
+sub checkDeltaFile
+{
+    my $self = shift;
+
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $stryFileList,
+        $oPriorManifest,
+        $lTimeBegin,
+    ) =
+        logDebugParam
+        (
+            __PACKAGE__ . '->checkDeltaFile', \@_,
+            {name => 'stryFileList'},
+            {name => 'oPriorManifest', required => false},
+            {name => 'lTimeBegin', required => false},
+        );
+
+    my $bDelta = false;
+
+    # Loop though all files
+    foreach my $strName (@{$stryFileList})
+    {
+        # If $lTimeBegin is defined, then this is not an aborted manifest so check if modification time is in the future (in this
+        # backup OR the last backup) then enable delta and exit
+        if (defined($lTimeBegin) &&
+            ($self->numericGet(MANIFEST_SECTION_TARGET_FILE, $strName, MANIFEST_SUBKEY_TIMESTAMP) > $lTimeBegin ||
+            (defined($oPriorManifest) &&
+             $oPriorManifest->test(MANIFEST_SECTION_TARGET_FILE, $strName, MANIFEST_SUBKEY_FUTURE, 'y'))))
+        {
+            &log(WARN, "file $strName has timestamp in the future, enabling delta checksum");
+            $bDelta = true;
+            last;
+        }
+
+        # If the time on the file is earlier than the last manifest time or the size is different but the timestamp is the
+        # same, then enable delta and exit
+        if (defined($oPriorManifest) && $oPriorManifest->test(MANIFEST_SECTION_TARGET_FILE, $strName) &&
+            ($self->numericGet(MANIFEST_SECTION_TARGET_FILE, $strName, MANIFEST_SUBKEY_TIMESTAMP) <
+            $oPriorManifest->numericGet(MANIFEST_SECTION_TARGET_FILE, $strName, MANIFEST_SUBKEY_TIMESTAMP) ||
+            ($self->numericGet(MANIFEST_SECTION_TARGET_FILE, $strName, MANIFEST_SUBKEY_SIZE) !=
+            $oPriorManifest->numericGet(MANIFEST_SECTION_TARGET_FILE, $strName, MANIFEST_SUBKEY_SIZE) &&
+            $self->numericGet(MANIFEST_SECTION_TARGET_FILE, $strName, MANIFEST_SUBKEY_TIMESTAMP) ==
+            $oPriorManifest->numericGet(MANIFEST_SECTION_TARGET_FILE, $strName, MANIFEST_SUBKEY_TIMESTAMP))))
+        {
+            &log(WARN, "file $strName timestamp in the past or size changed but timestamp did not, enabling delta checksum");
+            $bDelta = true;
+            last;
+        }
+    }
+
+    # Return from function and log return values if any
+    return logDebugReturn
+    (
+        $strOperation,
+        {name => 'bDelta', value => $bDelta, trace => true},
+    );
+}
+
+####################################################################################################################################
+# copyPriorChecksum
+#
```

Comment:
What is the purpose of this function?  It causes code churn (which needs review) but is only called from one place.